Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CoordinatedGraphicsScene.cpp

```diff
@@ -81,29 +67,31 @@ void CoordinatedGraphicsScene::paintToCurrentGLContext(const TransformationMatri
 
     bool sceneHasRunningAnimations = currentRootLayer->applyAnimationsRecursively(MonotonicTime::now());
 
+    bool didChangeClipRect = false;
     FloatRoundedRect actualClipRect(clipRect);
-#if ENABLE(WPE_PLATFORM) || PLATFORM(GTK)
-    if (m_propagateDamage != Damage::ShouldPropagate::No) {
-        WTFBeginSignpost(this, CollectDamage);
-        currentRootLayer->collectDamage(*m_textureMapper);
-        WTFEndSignpost(this, CollectDamage);
-
-        WebCore::Damage boundsDamage;
-        const auto& frameDamage = ([this, &boundsDamage, &unifyDamagedRegions]() -> const WebCore::Damage& {
-            const auto& damage = lastDamage();
-            if (m_propagateDamage != Damage::ShouldPropagate::No && !damage.isInvalid()) {
-                if (unifyDamagedRegions) {
-                    boundsDamage.add(damage.bounds());
-                    return boundsDamage;
-                }
-                return damage;
+#if ENABLE(DAMAGE_TRACKING)
+    if (m_client && m_damagePropagation != Damage::Propagation::None) {
+        Damage frameDamage;
+        if (sceneHasRunningAnimations) {
+            // When running animations for now we need to damage the whole frame.
+            frameDamage.add(clipRect);
+        } else {
+            WTFBeginSignpost(this, CollectDamage);
+            currentRootLayer->collectDamage(*m_textureMapper, frameDamage);
+            WTFEndSignpost(this, CollectDamage);
+
+            ASSERT(!frameDamage.isInvalid());
```

Comment:
Aye, we are calling `TextureMapperLayer::collectDamage()` always with a fresh `Damage`, and that itself won't invalidate the instance. Having the assertion here documents this expectation and then the check inside the `if` in the next line is fine to remove.